types/jest/index.d.ts

```diff
@@ -779,23 +819,35 @@ declare namespace jest {
         toHaveBeenCalledTimes(expected: number): R;
         /**
          * Ensure that a mock function is called with specific arguments.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        toHaveBeenCalledWith(...params: any[]): R;
+        toHaveBeenCalledWith<E extends any[]>(...params: E): R;
         /**
          * Ensure that a mock function is called with specific arguments on an Nth call.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        toHaveBeenNthCalledWith(nthCall: number, ...params: any[]): R;
+        toHaveBeenNthCalledWith<E extends any[]>(nthCall: number, ...params: E): R;
         /**
          * If you have a mock function, you can use `.toHaveBeenLastCalledWith`
          * to test what arguments it was last called with.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        toHaveBeenLastCalledWith(...params: any[]): R;
+        toHaveBeenLastCalledWith<E extends any[]>(...params: E): R;
         /**
          * Use to test the specific value that a mock function last returned.
          * If the last call to the mock function threw an error, then this matcher will fail
          * no matter what value you provided as the expected return value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toHaveLastReturnedWith(expected: any): R;
+        toHaveLastReturnedWith<E>(expected: E): R;
```

Comment:
```suggestion
        toHaveLastReturnedWith<E = any>(expected: E): R;
```

Explicit fallback rather than implicit